.github/workflows/build.yml

```diff
@@ -0,0 +1,168 @@
+name: Build cht-core and test against node versions
+
+on: [push, pull_request]
+
+env:
+  COUCH_URL: http://admin:pass@localhost:5984/medic-test
+  COUCH_NODE_NAME: nonode@nohost
+  BUILDS_SERVER: _couch/builds_testing
+  STAGING_SERVER: _couch/builds
+  UPLOAD_URL: ${{ secrets.UPLOAD_URL }}
+
+
+jobs:
+  build:
+    name: Compile the app 
+    runs-on: ubuntu-18.04
+
+    steps: 
+    # - name: Login to Docker Hub
+    #   uses: docker/login-action@v1
+    #   with:
+    #     username: ${{ secrets.DOCKERHUB_USERNAME }}
+    #     password: ${{ secrets.DOCKERHUB_TOKEN }}
+    - name: Set UPLOARD_URL var
+      run: |
+        if [[ -z "$UPLOAD_URL" ]]; then
+          echo "UPLOAD_URL=https://staging.dev.medicmobile.org" >> $GITHUB_ENV
+          echo "BUILDS_SERVER=_couch/builds_external" >> $GITHUB_ENV
+          echo "STAGING_SERVER=_couch/builds_external" >> $GITHUB_ENV
+        fi
+    - name: Get branch name
+      uses: nelonoel/branch-name@v1.0.1
```

Comment:
Since this is the only 3rd party `uses` we have, do we want to consider re-implementing the code in our own repo?  It's [only 10 lines](https://github.com/nelonoel/branch-name/blob/master/index.js) and would remove external dependencies for cht-core GH actions.  [MIT](https://github.com/nelonoel/branch-name/blob/master/LICENSE)!

I'll ask about the larger dependency/security posture in a comment too.

Reply:
I think you're right. This is easy enough to implement locally and have done so in another workflow. I was just being a bit lazy and looking for something that just works without much brain power. I'll roll this into a reusable step.

Reply:
The 10 lines isn't what is actually used. That seems to be an old reference. The action runs this [bit](https://github.com/nelonoel/branch-name/blob/master/dist/index.js).